### PR TITLE
Use dummy ACME cert initially; Reload nginx immediately before ACME issuance

### LINKED ISSUE #931: Don't listen on port 80 if acme is disabled
hello @jamilbk I installed the latest version of firezone that you fixed this issue, I tottaly uninstalled with the script and installed firezone. I configured some important options, and I still having a problem with accessing the admin panel and nginx is showing the same error as before. 

Screenshot from 2022-08-21 00-50-18

Here is my firezone config file:

```
{
  "firezone": {
    "external_url": "https://mydomain.com",
    "admin_email": "admin@mydomain.com",
    "max_devices_per_user": 10,
    "allow_unprivileged_device_management": true,
    "allow_unprivileged_device_configuration": true,
    "config_directory": "/etc/firezone",
    "install_directory": "/opt/firezone",
    "app_directory": "/opt/firezone/embedded/service/firezone",
    "log_directory": "/var/log/firezone",
    "var_directory": "/var/opt/firezone",
    "user": "firezone",
    "group": "firezone",
    "egress_interface": "enp3s0",
    "fips_enabled": false,
    "logging": {
      "enabled": true
    },
    "install_path": "/opt/firezone",
    "sysvinit_id": "SUP",
    "authentication": {
      "local": {
        "enabled": true
      },
      "auto_create_oidc_users": true,
      "disable_vpn_on_oidc_error": false,
      "oidc": {
 
      }
    },
    "phoenix": {
      "external_trusted_proxies": [
 
      ],
      "private_clients": [
 
      ],
      "enabled": true,
      "listen_address": "127.0.0.1",
      "port": 13000,
      "log_directory": "/var/log/firezone/phoenix",
      "log_rotation": {
        "file_maxbytes": 104857600,
        "num_to_keep": 10
      },
      "crash_detection": {
        "enabled": true
      },
      "secure_cookies": true
    },
    "nginx": {
      "enabled": true,
      "ssl_port": 3001,
      "directory": "/var/opt/firezone/nginx/etc",
      "log_directory": "/var/log/firezone/nginx",
      "log_rotation": {
        "file_maxbytes": 104857600,
        "num_to_keep": 10
      },
      "log_x_forwarded_for": true,
      "hsts_header": {
        "enabled": true,
        "include_subdomains": true,
        "max_age": 31536000
      },
      "ipv6": true,
      "redirect_to_canonical": false,
      "cache": {
        "enabled": false,
        "directory": "/var/opt/firezone/nginx/cache"
      },
      "user": "firezone",
      "group": "firezone",
      "dir": "/var/opt/firezone/nginx/etc",
      "log_dir": "/var/log/firezone/nginx",
      "pid": "/var/opt/firezone/nginx/etc/nginx.pid",
      "daemon_disable": true,
      "gzip": "on",
      "gzip_static": "off",
      "gzip_http_version": "1.0",
      "gzip_comp_level": "2",
      "gzip_proxied": "any",
      "gzip_vary": "off",
      "gzip_buffers": null,
      "gzip_types": [
        "text/plain",
        "text/css",
        "application/x-javascript",
        "text/xml",
        "application/xml",
        "application/rss+xml",
        "application/atom+xml",
        "text/javascript",
        "application/javascript",
        "application/json"
      ],
      "gzip_min_length": 1000,
      "gzip_disable": "MSIE [1-6]\\.",
      "keepalive": "on",
      "keepalive_timeout": 65,
      "worker_processes": 4,
      "worker_connections": 1024,
      "worker_rlimit_nofile": null,
      "multi_accept": true,
      "event": "epoll",
      "server_tokens": null,
      "server_names_hash_bucket_size": 64,
      "sendfile": "on",
      "access_log_options": null,
      "error_log_options": null,
      "disable_access_log": false,
      "types_hash_max_size": 2048,
      "types_hash_bucket_size": 64,
      "proxy_read_timeout": null,
      "client_body_buffer_size": null,
      "client_max_body_size": "250m",
      "default": {
        "modules": [
 
        ]
      },
      "enable_rate_limiting": true,
      "rate_limiting_zone_name": "firezone",
      "rate_limiting_backoff": "10m",
      "rate_limit": "10r/s"
    },
    "postgresql": {
      "enabled": true,
      "username": "firezone",
      "data_directory": "/var/opt/firezone/postgresql/13.3/data",
      "log_directory": "/var/log/firezone/postgresql",
      "log_rotation": {
        "file_maxbytes": 104857600,
        "num_to_keep": 10
      },
      "checkpoint_completion_target": 0.5,
      "checkpoint_segments": 3,
      "checkpoint_timeout": "5min",
      "checkpoint_warning": "30s",
      "effective_cache_size": "128MB",
      "listen_address": "127.0.0.1",
      "max_connections": 350,
      "md5_auth_cidr_addresses": [
        "127.0.0.1/32",
        "::1/128"
      ],
      "port": 15432,
      "shared_buffers": "951MB",
      "shmmax": 17179869184,
      "shmall": 4194304,
      "work_mem": "8MB"
    },
    "database": {
      "user": "firezone",
      "name": "firezone",
      "host": "127.0.0.1",
      "port": 15432,
      "ssl": false,
      "ssl_opts": {
 
      },
      "parameters": {
 
      },
      "pool": 10,
      "extensions": {
        "plpgsql": true,
        "pg_trgm": true
      },
      "create_user": true,
      "create_db": true
    },
    "wireguard": {
      "enabled": true,
      "log_directory": "/var/log/firezone/wireguard",
      "log_rotation": {
        "file_maxbytes": 104857600,
        "num_to_keep": 10
      },
      "interface_name": "wg-firezone",
      "port": 51820,
      "mtu": 1280,
      "endpoint": "myIP",
      "allowed_ips": "0.0.0.0/0, ::/0",
      "dns": "1.1.1.1, 1.0.0.1",
      "persistent_keepalive": 0,
      "ipv4": {
        "enabled": true,
        "masquerade": true,
        "network": "10.3.2.0/24",
        "address": "10.3.2.1"
      },
      "ipv6": {
        "enabled": true,
        "masquerade": true,
        "network": "fd00::3:2:0/120",
        "address": "fd00::3:2:1"
      }
    },
    "runit": {
      "svlogd_bin": "/opt/firezone/embedded/bin/svlogd"
    },
    "ssl": {
      "directory": "/var/opt/firezone/ssl",
      "email_address": "admin@acso.gov.et",
      "acme": {
        "enabled": false,
        "server": "letsencrypt",
        "keylength": "ec-256"
      },
      "certificate": "/etc/letsencrypt/live/mydomain/fullchain.pem",
      "certificate_key": "/etc/letsencrypt/live/mydomain/privkey.pem",
      "ssl_dhparam": "/var/opt/firezone/ssl/ca/dhparams.pem",
      "country_name": "US",
      "state_name": "CA",
      "locality_name": "San Francisco",
      "company_name": "My Company",
      "organizational_unit_name": "Operations",
      "ciphers": "ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:DHE-DSS-AES128-GCM-SHA256:kEDH+AESGCM:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA:ECDHE-ECDSA-AES256-SHA:DHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA:DHE-DSS-AES128-SHA256:DHE-RSA-AES256-SHA256:DHE-DSS-AES256-SHA:DHE-RSA-AES256-SHA:AES128-GCM-SHA256:AES256-GCM-SHA384:AES128-SHA:AES256-SHA:AES:CAMELLIA:DES-CBC3-SHA:!aNULL:!eNULL:!EXPORT:!DES:!RC4:!MD5:!PSK:!aECDH:!EDH-DSS-DES-CBC3-SHA:!EDH-RSA-DES-CBC3-SHA:!KRB5-DES-CBC3-SHA",
      "fips_ciphers": "FIPS@STRENGTH:!aNULL:!eNULL",
      "protocols": "TLSv1 TLSv1.1 TLSv1.2",
      "session_cache": "shared:SSL:4m",
      "session_timeout": "5m"
    },
    "robots_allow": "/",
    "robots_disallow": null,
    "outbound_email": {
      "from": null,
      "provider": null,
      "configs": {
        "smtp": {
          "relay": "smtp.example.com",
          "port": 587,
          "username": "",
          "password": "",
          "ssl": true,
          "tls": "always",
          "auth": "always",
          "no_mx_lookup": false,
          "retries": 2
        },
        "mailgun": {
          "apikey": null,
          "domain": null
        },
        "mandrill": {
          "api_key": null
        },
        "sendgrid": {
          "api_key": null
        },
        "post_mark": {
          "api_key": null
        },
        "sendmail": {
          "cmd_path": "/usr/bin/sendmail",
          "cmd_args": "-N delay,failure,success"
        }
      }
    },
    "telemetry": {
      "enabled": true
    },
    "connectivity_checks": {
      "enabled": true,
      "interval": 3600
    },
    "telemetry_id": "4550d3a7-0812-40e6-a34f-57ec5aa0407f",
    "guardian_secret_key": "QIDVpe3W+jqSzs7Nc4Bdjg4VdcftN5KtzwMKaMBghkgrVD6TAXi650fl9SF2I9hS",
    "secret_key_base": "Hz4wYR1E6kJIHXZWHlCTUKtxm+QeyYIW0LNIqhBdNhpoUKLDW4PJ/FMwrx7xqLNX",
    "live_view_signing_salt": "9lGo+opUyAGkFC7/VAgTJnSOibGbIAiJ",
    "cookie_signing_salt": "MI/WgGub",
    "cookie_encryption_salt": "NFNKl1eG",
    "database_encryption_key": "+W2xlMb3KNKoCty82GXUBk/jp9/4ukgOfaRW7YA/2N8=",
    "default_admin_password": "7BnnlOhAf3c4KaEl"
  }
} 
```

_Originally posted by @elhananjair in https://github.com/firezone/firezone/issues/920#issuecomment-1221416451_

### PR DESCRIPTION
**Pull request by jamilbk**
* Reload nginx config just before ACME issuance to ensure it's listening on port 80
* Use dummy ACME cert when writing `phoenix.nginx.conf` so that nginx starts without erroring

Fixes #931

### CODE REVIEW

**Comment by jamilbk** [comment]
@conectado Ok so this is a bigger issue than I originally thought. There's somewhat of a dependency cycle between Nginx restarts and acme config.

If ACME is enabled:

1. phoenix.nginx.conf gets written with the ACME ssl cert path
1. redirect.conf is enabled
1. Nginx is restarted, but fails to start because the ACME cert isn't there yet
1. ACME initialization begins, but can't verify because Nginx isn't running

Will have to think a little more about how to get around this.

**Comment by conectado** [comment]
> @conectado Ok so this is a bigger issue than I originally thought. There's somewhat of a dependency cycle between Nginx restarts and acme config.
> 
> If ACME is enabled:
> 
>     1. phoenix.nginx.conf gets written with the ACME ssl cert path
> 
>     2. redirect.conf is enabled
> 
>     3. Nginx is restarted, but fails to start because the ACME cert isn't there yet
> 
>     4. ACME initialization begins, but can't verify because Nginx isn't running
> 
> 
> Will have to think a little more about how to get around this.

We might need to comment out `sites-enabled` until we have the certs.

**Comment by conectado** [comment]
> We might need to comment out `sites-enabled` until we have the certs.

Maybe a faster temporary workaround would be to mock the certs with the self-generated certs.

Then we force the cert regeneration.

File: omnibus/cookbooks/firezone/recipes/nginx.rb [hunk 1]
```diff
@@ -64,7 +64,8 @@
     server_name: URI.parse(node['firezone']['external_url']).host,
     acme_www_root: "#{node['firezone']['var_directory']}/nginx/acme_root",
     rate_limiting_zone_name: node['firezone']['nginx']['rate_limiting_zone_name'],
-    ipv6: node['firezone']['nginx']['ipv6']
+    ipv6: node['firezone']['nginx']['ipv6'],
+    acme: { 'enabled' => node['firezone']['ssl']['acme']['enabled'] }
```
Comment (conectado) [inline omnibus/cookbooks/firezone/recipes/nginx.rb]:
```suggestion
    acme: node['firezone']['ssl']['acme']
```

**Review by conectado — DISMISSED** [review]
Looks good, left a minor suggestion.

We just need to make sure that the acme cert is generated, since it's just the cert in the install path it probably will be but otherwise we might need to pass a `--force` flag to `acme.sh`.

**Review by conectado — DISMISSED** [review]
lgtm!

**Comment by jamilbk** [comment]
@conectado Last piece of the puzzle was to reload nginx just before the issue step. Working fine now.

**Pull request merged**